juriscraper/opinions/united_states/state/nh.py

```diff
@@ -15,47 +15,145 @@
 """
 
+import re
+from datetime import datetime
+from typing import Dict, List
+from urllib.parse import urlencode, urljoin
+
+from juriscraper.AbstractSite import logger
 from juriscraper.OpinionSiteLinear import OpinionSiteLinear
 
 
 class Site(OpinionSiteLinear):
+    # document_purpose = 1331 -> Supreme Court Opinion
+    base_filter = "{}@field_document_purpose|=|1331"
+    year_to_filter = {
+        2024: "@field_document_subcategory|=|2316",
+        2023: "@field_document_subcategory|=|2256",
+        2022: "@field_document_subcategory|=|2091",
+        2021: "@field_tags|CONTAINS|1206~field_entity_tags|CONTAINS|1206",
+        2020: "@field_tags|CONTAINS|1366~field_entity_tags|CONTAINS|1366",
+        2019: "@field_tags|CONTAINS|1416~field_entity_tags|CONTAINS|1416",
+        2018: "@field_document_subcategory|=|1601",
+        2017: "@field_document_subcategory|=|1596",
+        2016: "@field_document_subcategory|=|1591",
+        2015: "@field_document_subcategory|=|1586",
+    }
+    # there is data since 2002, but we would need to
+    # collect all subcategory or tag values
+    start_year = 2015
+    end_year = datetime.today().year - 1
+
+    params = {
+        "iterate_nodes": "true",
+        "q": "",
+        "sort": "field_date_posted|desc|ALLOW_NULLS",
+        "filter_mode": "exclusive",
+        "type": "document",
+        "page": "1",
+        "size": "25",
+    }
+    docket_regex = re.compile(r"\d{4}-\d{4}")
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.court_id = self.__module__
-        self.url = "https://www.courts.nh.gov/content/api/documents?iterate_nodes=true&q=%40field_document_subcategory%7C%3D%7C2256%40field_document_purpose%7C%3D%7C1331&sort=field_date_posted%7Cdesc%7CALLOW_NULLS&filter_mode=exclusive&page=1&size=25"
         self.status = "Published"
-
-    def _download(self):
-        if self.test_mode_enabled():
-            import json
-
-            return json.load(open(self.url))
-        return (
-            self.request["session"]
-            .get(
-                self.url,
-                headers=self.request["headers"],
-                verify=self.request["verify"],
-                timeout=60,
-            )
-            .json()
-        )
+        self.base_url = "https://www.courts.nh.gov/content/api/documents"
+        self.set_request_parameters()
+        self.needs_special_headers = True
+        self.make_backscrape_iterable(kwargs)
+        self.paginate = False
 
     def _process_html(self) -> None:
-        for case in self.html["data"]:
+        json_response = self.html
+
+        for case in json_response["data"]:
             url = case["fields"]["field_document_file"]["0"]["fields"]["uri"][
                 0
             ].split("//")[1]
+
             if "," not in case["title"]:
+                logger.info("Skipping case with title %s", case["title"])
```

Comment:
? what is happening here?

Reply:
Why do we need this skipper?  Looks like this opinion 

https://www.courts.nh.gov/sites/g/files/ehbemt471/files/documents/2024-05/2024026farrell.pdf

gets skipped because the CRM is missing a comma.

Reply:
This comes from the previous scraper. There was no comment explaining why, but I think it's because we get the docket / citation via comma separation

https://github.com/grossir/juriscraper/blob/c58581dbbccca13d2e4030d3c4a450f0b3f877d6/juriscraper/opinions/united_states/state/nh.py#L51-L52